src/basic/MAT_MAT_SHARED.hpp

```diff
@@ -0,0 +1,170 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-20, Lawrence Livermore National Security, LLC
+// and RAJA Performance Suite project contributors.
+// See the RAJAPerf/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+///
+/// Matrix matrix multiplication with shared memory
+///
+///
+
+#ifndef RAJAPerf_Basic_MAT_MAT_SHARED_HPP
+#define RAJAPerf_Basic_MAT_MAT_SHARED_HPP
+
+#include "RAJA/RAJA.hpp"
+#include "common/KernelBase.hpp"
+
+#define TL_SZ 16
+
+#define MAT_MAT_SHARED_DATA_SETUP                                              \
+  Real_ptr A = m_A;                                                            \
+  Real_ptr B = m_B;                                                            \
+  Real_ptr C = m_C;
+
+//Clang has trouble with the RAJA_TEAM_SHARED macros believing
+//its always compiling for the device
```

Comment:
```suggestion
/*
 When doing the device compile pass hipcc/clang will put in the device
 versions of the macros everywhere, in device functions, host device functions, 
 and host only functions. Then it will make sure that code is valid everywhere, 
 that's fine for device and host device functions, but it is not ok for host only 
 functions. Nvcc doesn't look at host only code when it does the device pass 
 so it doesn't see these kind of problems.
 */
```